catroid/src/main/java/org/catrobat/catroid/stage/StageListener.java

```diff
@@ -216,6 +219,49 @@ public void create() {
 		}
 	}
 
+	public void cloneSpriteAndAddToStage(Sprite cloneMe) {
+		Sprite copy = cloneMe.cloneForCloneBrick();
+		copy.resetSprite();
+		copy.look.createBrightnessContrastHueShader();
+		stage.addActor(copy.look);
+		copy.resume();
+		sprites.add(copy);
+		clonedSprites.add(copy);
+
+		Map<String, List<String>> scriptActions = new HashMap<>();
+		copy.createStartScriptActionSequenceAndPutToMap(scriptActions);
+		precomputeActionsForBroadcastEvents(scriptActions);
+		if (!copy.getLookDataList().isEmpty()) {
+			copy.look.setLookData(copy.getLookDataList().get(0));
+		}
+
+		copy.createWhenClonedAction();
+	}
+
+	public void removeClonedSpriteFromStage(Sprite sprite) {
+		if (!sprite.isClone) {
+			return;
+		}
+
+		Project currentProject = ProjectManager.getInstance().getCurrentProject();
+		DataContainer userVariables = currentProject.getDataContainer();
+		userVariables.removeVariableListForSprite(sprite);
+
+		BroadcastHandler.getScriptSpriteMapMap().remove(sprite);
```

Comment:
There's a typo in the existing code: please change getScriptSpriteMapMap --> getScriptSpriteMap